Neos.ContentRepository/Classes/Configuration/RecursiveDirectoryIteratorWithSorting.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Configuration;
+
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * The {@see \RecursiveDirectoryIterator} doesn't order the returned files.
+ *
+ * https://www.php.net/manual/en/class.recursivedirectoryiterator.php#120971
+ * > On Windows, you will get the files ordered by name. On Linux they are not ordered.
+ *
+ * To enforce a deterministic behavior, we wrap the RecursiveDirectoryIterator and sort the files level per level.
+ *
+ * This iterator is lazy, but even fetching one item will evaluate the inner RecursiveDirectoryIterator (one level of the directories)
+ *
+ * The sorting strategy is as follows:
+ *
+ * We sort directory by directory via the base name.
+ *
+ * This structure (unordered, so show how it might be scanned by the RecursiveDirectoryIterator)
+ *
+ * Content/
+ * ├─ Z.yaml
+ * ├─ A.yaml
+ * ├─ Columns/
+ * │  ├─ B.yaml
+ * │  ├─ A.yaml
+ *
+ * Will be sorted into
+ *
+ * Content/
+ * ├─ A.yaml
+ * ├─ Columns/
+ * │  ├─ A.yaml
+ * │  ├─ B.yaml
+ * ├─ Z.yaml
+ *
+ * so after applying an {@see \RecursiveIteratorIterator} one will get
+ *
+ * Content/A.yaml
+ * Content/Columns/A.yaml
+ * Content/Columns/B.yaml
+ * Content/Z.yaml
+ *
+ * @Flow\Proxy(false)
+ * @internal
+ */
+class RecursiveDirectoryIteratorWithSorting implements \RecursiveIterator
+{
+    private const KEY = 0;
+
+    private const FILE_INFO = 1;
+
+    private const CHILDREN = 2;
+
+    /**
+     * @var \RecursiveDirectoryIterator
+     */
+    private $innerRecursiveDirectoryIterator;
+
+    /**
+     * The initialized one level structure of the inner RecursiveDirectoryIterator
+     * @var array<string, array<int, mixed>
+     */
+    private array $files;
```

Comment:
Typed properties are not supported with PHP 7.3. See what the failing test says:

```
Running in context 'Testing'
syntax error, unexpected 'array' (T_ARRAY), expecting function (T_FUNCTION) or const (T_CONST)
Type: ParseError
  File:
Packages/Neos/Neos.ContentRepository/Classes/Configuration/RecursiveDirecto
        ryIteratorWithSorting.php
  Line: 68

  Type: Neos\Flow\Core\Booting\Exception\SubProcessException
  Code: 1355480641
  File: Packages/Framework/Neos.Flow/Classes/Core/Booting/Scripts.php
  Line: 712

Open Data/Logs/Exceptions/202307100850539d3861.txt for a full stack trace.
```

Reply:
wait but i thought php 7.4 was the oldest version we allow in Neos AND Flow? So flow has different minimal dependencies?

Reply:
seems i was wrong ^^

Reply:
https://docs.neos.io/guide/installation-development-setup/system-requirements#php-version-matrix – sad, but true. 😇